### ISSUE TITLE
Incorrect library name on Windows with layout=versioned

### ISSUE THREAD

**Issue by msmolensky** [boostorg/build#617]
Tried boost v 1.66 and 1.73. B2 generates wrong file name for versioned layout - address-model+architecture part is missing, e.g. boost_unit_test_framework-vc141-mt-gd-1_66.lib instead of boost_unit_test_framework-vc141-mt-gd-x64-1_66.lib

Adding <arch-and-model> to boost.jam tag_versioned rule fixes the problem. (sorry for not attaching the diff)

**Comment by mloskot** [comment]
Did you run `b2 address-model=64`?

**Comment by msmolensky** [comment]
Yes. Without this tag it is ignored in format-name (in common.jam), no
matter if you add address-model and/or architecture to command line or
Jamfile. I ran it in debugger.

 

Michael Smolensky

  _____

**Comment by mloskot** [comment]
Sorry, I don't understand what you are doing. Next time, simply paste full command you did run.

Today, I checked building 1.73 with VS 2019:


```
D:\tmp\boost_1_73_0> bootstrap.bat
D:\tmp\boost_1_73_0> b2 --layout=versioned variant=release,debug toolset=msvc-14.2 --with-test
...
D:\tmp\boost_1_73_0>ls -1 D:\tmp\boost_1_73_0\stage\lib
cmake
libboost_prg_exec_monitor-vc142-mt-gd-x32-1_73.lib
libboost_prg_exec_monitor-vc142-mt-gd-x64-1_73.lib
libboost_prg_exec_monitor-vc142-mt-x32-1_73.lib
libboost_prg_exec_monitor-vc142-mt-x64-1_73.lib
libboost_test_exec_monitor-vc142-mt-gd-x32-1_73.lib
libboost_test_exec_monitor-vc142-mt-gd-x64-1_73.lib
libboost_test_exec_monitor-vc142-mt-x32-1_73.lib
libboost_test_exec_monitor-vc142-mt-x64-1_73.lib
libboost_unit_test_framework-vc142-mt-gd-x32-1_73.lib
libboost_unit_test_framework-vc142-mt-gd-x64-1_73.lib
libboost_unit_test_framework-vc142-mt-x32-1_73.lib
libboost_unit_test_framework-vc142-mt-x64-1_73.lib
```

You can also be explicit about the address model, requesting both 32 and 64 bit targets:

```
b2 --layout=versioned address-model=32,64 variant=release,debug toolset=msvc-14.2
```

**Comment by msmolensky** [comment]
Sorry for not being clear. I am not building boost - it works fine. I made a
Jamfile and ran it with b2. The Jamfile referred to a boost library built
with versioned layout. The build failed because the library name was
generated incorrectly -x64 was missing. Unfortunately, I did it on a
different machine that I've submitted bug from so I could not copy command
line, error or Jamfile. The Jamfile was something like:

 

exe Sample

: ...

: ...

 <library>/boost//unit_test_framework

;

 

If it is still not clear, I will reproduce it locally and submit code sample
and error output

 

Michael Smolensky

  _____

**Comment by msmolensky** [comment]
Noticed that the original post was auto formatted.  Should be "Adding **\<arc-and-model>** to boost.jam tag_versioned rule fixes the problem"
